bad_connector_woocommerce/controllers/main.py

```diff
@@ -0,0 +1,79 @@
+import json
+import logging
+from urllib.parse import urlparse
+
+from werkzeug.exceptions import Forbidden
+
+from odoo import http
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class WooWebhook(http.Controller):
+    def _common_webhook_handler(self, access_token, model_name):
+        """Common handler for processing WooCommerce webhooks."""
+        job_options = {}
+        backend = (
+            request.env["woo.backend"]
+            .sudo()
+            .search(
+                [
+                    "|",
+                    "&",
+                    ("test_mode", "!=", True),
+                    ("access_token", "=", access_token),
+                    "&",
+                    ("test_access_token", "=", access_token),
+                    ("test_mode", "=", True),
+                ],
+                limit=1,
+            )
+        )
+        if not backend:
+            _logger.error(
+                "No WooCommerce backend found. Check your Access Token and try again"
+            )
+            raise Forbidden()
+        base_url = request.httprequest.base_url
+        parsed_url = urlparse(base_url)
+        base_url = f"{parsed_url.scheme}://{parsed_url.netloc}"
+        backend.write({"base_url": base_url})
```

Comment:
why this is needed?

Reply:
We don't need this. I will remove it and take the web.base.url